### PR TITLE
NAS-129466 / 24.10 / Fix failing tests because of apps removal

### PR DESCRIPTION
**Pull request by Qubad786**
## Context

We had some leftover usages of `chart.release` namespace which have been covered here and were also causing some tests to fail.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/pool_/dataset_details.py [hunk 1]
```diff
@@ -325,19 +325,7 @@ def build_details(self, mntinfo):
...
-            for path in filter(
-                lambda x: x.startswith('/mnt/') and 'ix-applications/' not in x,
-                paths
-            ):
-                results['app'].append({
-                    'name': app_name,
-                    'path': path,
-                    'mount_info': self.get_mount_info(path, mntinfo),
-                })
+        # FIXME: Add app details here, currently it would be an empty list
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/pool_/dataset_details.py:328]: Let's still return an "apps" key with an empty list so it doesn't break UI on the datasets page.

**Review by yocalebo — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/pool_/dataset_details.py (hunk 1 above)
Comment (Qubad786) [reply src/middlewared/middlewared/plugins/pool_/dataset_details.py:328]: We are doing that, it's already defined as empty above

Comment (Qubad786) [reply src/middlewared/middlewared/plugins/pool_/dataset_details.py:328]: @yocalebo here https://github.com/truenas/middleware/blob/fd8e3e12ff05adf5b4d7743696e9ee8a22bd8fa3/src/middlewared/middlewared/plugins/pool_/dataset_details.py#L271-L271

**Review by yocalebo — APPROVED** [review]
Thanks!

**Review by aiden3c — APPROVED** [review]

**Pull request merged**